### PR TITLE
[media] use languageLabel derived from language_id to display

### PR DESCRIPTION
**Pull request by PapillonMcGill**
## Brief summary of changes

- [ ] Have you updated related documentation?

#### Testing instructions (if applicable)

- go to media module, check if language of document is properly displayed in table
- edit a document with a language, current language should be displayed in the edit window.
- modify the language and check that change are properly reflected in table

#### Link(s) to related issue(s)

* Resolves #  (Reference the issue this fixes, if any.)
https://github.com/aces/Loris/issues/6312

### CODE REVIEW

File: modules/media/ajax/FileUpload.php [hunk 1]
```diff
@@ -300,7 +300,7 @@ function getUploadFields()
             "comments, " .
             "file_name as fileName, " .
             "hide_file as hideFile, " .
-            "language_id as language," .
+            "language_id as language_id," .
```
Comment (ridz1208) [inline modules/media/ajax/FileUpload.php:303]: you can remove the as since `X as X` is the same as just `X`

File: modules/media/jsx/editForm.js [hunk 2]
```diff
@@ -40,6 +40,7 @@ class MediaEditForm extends Component {
           dateTaken: data.mediaData.dateTaken,
           comments: data.mediaData.comments,
           hideFile: data.mediaData.hideFile,
+          language_id: data.mediaData.language_id,
```
Comment (ridz1208) [inline modules/media/jsx/editForm.js:43]: After reviewing all the changes, it seems like the only change needed here is simple adding this line. and if you undo all the renaming of variables, then this line becomes

```suggestion
          language: data.mediaData.language,
```

I just tested it out and it works. Is there any justification for all the other changes ?

Comment (PapillonMcGill) [reply modules/media/jsx/editForm.js:43]: Yes, there was 2 defects in the module, all related to language. The fact that the label is in a separate table witrh an FK in the media table. All this was missed during reactification.

Comment (ridz1208) [reply modules/media/jsx/editForm.js:43]: what's the other defect ? from what I could tell the only issue was that when loading the edit form, the language was not displayed.

Comment (PapillonMcGill) [reply modules/media/jsx/editForm.js:43]: `language` is for the list of possible language and `language_id` the one that the documents is tag with.

Comment (ridz1208) [reply modules/media/jsx/editForm.js:43]: I'm not sure I understand. After removing all changes and just changing this one line everything is functional for me.

I'll leave the decision to other reviewers.

File: modules/media/php/mediafileprovisioner.class.inc [hunk 3]
```diff
@@ -38,22 +38,23 @@ class MediaFileProvisioner extends \LORIS\Data\Provisioners\DBRowProvisioner
...
+                m.uploaded_by as uploadedBy,
+                m.date_taken as dateTaken,
+                substring(m.comments, 1, 50) as comments,
+                m.last_modified as lastModified,
+                m.file_type as fileType,
+                s.CandID as candidateId,
+                s.ID as sessionId,
+                m.hide_file as hideFile,
+                m.id,
+                m.language_id as language
```
Comment (ridz1208) [inline modules/media/php/mediafileprovisioner.class.inc:57]: language label is already being queried on line 43.

why is this queried here ? and why use `language` when everywhere else changed to either `language_id` or `languageLabel` ?

Comment (PapillonMcGill) [reply modules/media/php/mediafileprovisioner.class.inc:57]: That line might be extra. I had trouble finding the exact place to correct for the edit window.

File: modules/media/jsx/editForm.js (hunk 2 above)
Comment (PapillonMcGill) [reply modules/media/jsx/editForm.js:43]: have you tried to edit an document meta-data without the other change? 2nd and 3rd bullet of how to test in the description.

Comment (ridz1208) [reply modules/media/jsx/editForm.js:43]: yes, give it a try.

File: modules/media/php/mediafileprovisioner.class.inc [hunk 4]
```diff
@@ -38,22 +38,22 @@ class MediaFileProvisioner extends \LORIS\Data\Provisioners\DBRowProvisioner
     {
         parent::__construct(
             "SELECT m.file_name as fileName,
-                    c.PSCID as pscid,
```
Comment (driusan) [inline modules/media/php/mediafileprovisioner.class.inc:41]: This shouldn't be making whitespace changes

Comment (PapillonMcGill) [reply modules/media/php/mediafileprovisioner.class.inc:41]: I know but ended up with missalign lines when I added a line here (that was later removed). I had to replace lots of tab with space.

File: modules/media/jsx/editForm.js (hunk 2 above)
Comment (ridz1208) [reply modules/media/jsx/editForm.js:43]: @PapillonMcGill have you had a chance to give it a try with only this line ?

File: modules/media/jsx/editForm.js [hunk 5]
```diff
@@ -136,6 +137,7 @@ class MediaEditForm extends Component {
           />
           <SelectElement
             name='language'
+            id='language_id'
```
Comment (PapillonMcGill) [inline modules/media/jsx/editForm.js:140]: @ridz1208 This line is required to have current value shown in the edit windows

File: modules/media/jsx/mediaIndex.js [hunk 6]
```diff
@@ -159,7 +159,7 @@ class MediaIndex extends Component {
         options: options.visits,
       }},
       {label: 'Language', show: true, filter: {
-        name: 'language',
+        name: 'languageLabel',
```
Comment (PapillonMcGill) [inline modules/media/jsx/mediaIndex.js:162]: @ridz1208 removing this line, the current value is not shown in the edit form.

File: modules/media/jsx/editForm.js (hunk 2 above)
Comment (PapillonMcGill) [reply modules/media/jsx/editForm.js]: @ridz1208 You're right, minimal change required. Not sure what was the cause but with other change, the modal windows closed automatically when I click on the OK button after update during initial test. While retrying, the modal windows always stayed after clicking on the Ok button after update.

**Comment by driusan** [comment]
@racostas can you confirm that this fixes the issue you reported?

**Comment by racostas** [comment]
> @racostas can you confirm that this fixes the issue you reported?

It seems to be working fine. Tested locally.

The following is very minor:
Found that once the file is loaded (the user been in the tab `Upload`), if the user goes to the tab `Browse` the new entry doesn't appear until manually refreshed the page (F5). The concern is when not knowledgeable users are doing the data entry.

File: modules/media/jsx/editForm.js (hunk 5 above)
Comment (ridz1208) [inline modules/media/jsx/editForm.js:140]: @PapillonMcGill @driusan this still seems to be an unnecessary change (removing it does not impact functionality) but since the PR fixes the issue I won't block it over this
